test/e2e/api/events/map-change-event.test.js

```diff
@@ -0,0 +1,101 @@
+import { test, expect, chromium } from '@playwright/test';
+
+test.describe('Map change event are only fired when layers/extents are checked or unchecked in the layer menu ', () => {
+  let page;
+  let context;
+  test.beforeAll(async () => {
+    context = await chromium.launchPersistentContext('', { headless: false });
+    page =
+      context.pages().find((page) => page.url() === 'about:blank') ||
+      (await context.newPage());
+    await page.goto('events/map-change-event.html');
+    await page.waitForTimeout(1000);
+  });
+
+  test.afterAll(async function () {
+    await context.close();
+  });
+
+  test('Map change event for layers work', async () => {
+    let layerClicked = 0;
+    page.on('console', (msg) => {
+      if (msg.text() === 'Layer clicked') {
+        layerClicked++;
+      }
+    });
+
+    // check and uncheck layers in the DOM
```

Comment:
update all comments so we know in the future (if we ever debug this) which actions are supposed to trigger the map-change and which aren't